### PR TITLE
foreman_host: Replace the "name" attribute with "fqdn" (computed) and "shortname" (required)

### PR DESCRIPTION
**Pull request by bitkeks**
As mentioned in #116 the "name" field is problematic in Foreman. The setting "append_domain_name_for_hosts" converts a short hostname into an FQDN. This leads to "inconsistent plan" errors in Terraform. 

In [earlier versions](https://github.com/terraform-coop/terraform-provider-foreman/commit/2fa9ff24ae723a3f21197ed15c32ef43415b67ee#diff-1c2c1e4478457d1014e760b4b65a9a558e986d83eeeb398b77148dd7721598dbL275), a custom unmarshaller removed the domain part from the name attribute:
``` 
if fh.DomainName != "" && strings.Contains(fh.ForemanObject.Name, fh.DomainName) {
    fh.ForemanObject.Name = strings.Replace(fh.ForemanObject.Name, "."+fh.DomainName, "", 1)
}
``` 
In this case, Foreman always used the FQDN internally, but Terraform received only the shortname in return and had no issues. Since this function is removed, the conflict appears.

For expected behaviour and consistency I propose a switch to "fqdn" and "shortname" instead of the "name" attribute. This PR is the implementation.

The result will look like this:

* `name`: Read-only, used in Foreman API as a unique name for a host. Can be either the short name or the expanded FQDN, depending on the respective Foreman setting.
* `shortname`: The "new" "name" argument that you should use in Terraform for your hosts. In combination with the `domain_id` Foreman knows what the domain part should be.
* `fqdn`: Read-only, only exists in the Terraform provider. Is constructed from the name received from Foreman. Can be used as a consistent variable value in other parts of the Terraform module files and blocks.

### CODE REVIEW

File: foreman/api/host.go [hunk 1]
```diff
@@ -427,10 +451,35 @@ func (c *Client) readComputeAttributes(ctx context.Context, id int) (map[string]
...
+
+func constructShortname(host *foremanHostDecode) error {
+	log.Tracef("foreman/api/host.go#constructShortname")
+
+	// Construct shortname from 'name'
+	if host.Shortname == "" {
+		before, after, found := strings.Cut(host.Name, ".")
+
+		// If no dot is found, there's a serious error with the Foreman name attribute for this host
+		if !found {
```
Comment (bitkeks) [inline foreman/api/host.go:469]: This might not be an error if the append_domain setting is false?

Related to the comment in https://github.com/terraform-coop/terraform-provider-foreman/pull/120#discussion_r1226496770, we should merge the two PRs and then use the readSetting to handle this case.

Comment (bitkeks) [reply foreman/api/host.go:477]: Refactored and resolved in a1042160159aa2b1eee652aaabd2b9c2ce82cb9d

File: foreman/resource_foreman_host.go [hunk 2]
```diff
@@ -298,7 +300,28 @@ func resourceForemanHost() *schema.Resource {
...
+				newMap := expandComputeAttributes(newVal.(string))
+
+				err := mergo.Merge(&oldMap, newMap, mergo.WithOverride)
+
+				if err != nil {
+					log.Printf("[ERROR]: Could not merge defined and existing compute attributes, [%v]", err)
+				}
+
+				d.SetNew("compute_attributes", flattenComputeAttributes(oldMap))
+				return nil
```
Comment (lhw) [inline foreman/resource_foreman_host.go:318]: Why did you move this out of the function into the attribute here? That doesn't really help the visibility

File: foreman/resource_foreman_host.go [hunk 3]
```diff
@@ -298,7 +300,28 @@ func resourceForemanHost() *schema.Resource {
...
+			},
+			// customdiff.ComputedIf("name", func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+			// 	old, new := d.GetChange("name")
+			// 	return strings.HasPrefix(new.(string), old.(string))
+			// }),
```
[the remark this answers is not in the document]
Comment (bitkeks) [reply foreman/resource_foreman_host.go]: Old code, removed :+1: 
Some times it was necessary to keep the code in, but comment out, to not lose track of already tested changes.

**Review by lhw — APPROVED** [review]
